File: src/main/java/com/belum/apitemplate/controllers/advice/ApiRequest.java
package com.belum.apitemplate.controllers.advice;

import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import lombok.Getter;
import lombok.ToString;

import java.time.ZonedDateTime;
import java.time.format.DateTimeFormatter;
import java.time.format.FormatStyle;

@JsonPropertyOrder({"code", "message", "timeStamp"})
@Getter
@ToString
public final class ApiRequest {
    private final String code;
    private final String message;
    private final String timeStamp;

    public ApiRequest(String code, String message) {
        this.code = code;
        this.message = message;
        this.timeStamp = ZonedDateTime.now().format(DateTimeFormatter.ofLocalizedDateTime(FormatStyle.LONG));
    }
}

File: src/main/kotlin/com/bemsa/scraper/controllers/advice/ApiRequest.kt
package com.bemsa.scraper.controllers.advice

import com.fasterxml.jackson.annotation.JsonPropertyOrder

import java.time.ZonedDateTime
import java.time.format.DateTimeFormatter
import java.time.format.FormatStyle

@JsonPropertyOrder("code", "message", "timeStamp")
data class ApiRequest(
    private val code: String,
    private val message: String,
    private val timeStamp: String
) {
    constructor(code: String, message: String) : this(code, message, ZonedDateTime.now().format(DateTimeFormatter.ofLocalizedDateTime(FormatStyle.LONG)))
}
